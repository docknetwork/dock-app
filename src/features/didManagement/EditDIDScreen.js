import React, {useCallback} from 'react';
import {
  BackButton,
  Box,
  Header,
  Input,
  LoadingButton,
  NBox,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {navigateBack} from '../../core/navigation';
import {translate} from '../../locales';
import {FormControl, ScrollView, Stack} from 'native-base';
import {addTestId} from '../../core/automation-utils';
export function EditDIDScreen() {
  const onChange = useCallback(() => {}, []);
  return (
    <ScreenContainer testID="CreateNewDIDScreen">
      <Header>
        <Box
          marginLeft={1}
          marginRight={22}
          flexDirection="row"
          alignItems="center">
          <NBox width={'80px'}>
            <BackButton
              {...addTestId('EditDIDScreenGoBack')}
              onPress={navigateBack}
            />
          </NBox>
          <NBox
            flex={1}
            width="100%"
            alignContent="center"
            alignItems="center"
            pl={15}>
            <Typography variant="h3">
              {translate('didManagement.edit_did')}
            </Typography>
          </NBox>
          <NBox width="80px" alignItems="flex-end" />
        </Box>
      </Header>
      <ScrollView marginLeft={5} marginRight={5}>
        <FormControl>
          <Stack mt={7}>
            <FormControl.Label>
              {translate('didManagement.did_name')}
            </FormControl.Label>
            <Input
              {...addTestId('EditDIDScreenGoBack')}
              value={''}
              onChangeText={onChange('word1')}
              autoCapitalize="none"
            />
          </Stack>
        </FormControl>
      </ScrollView>
      <NBox mx={7}>
        <LoadingButton
          {...addTestId('EditDIDScreenSave')}
          full
          testID="save-btn"
          mb={70}
          onPress={null}
          isDisabled={false}>
          {translate('didManagement.save')}
        </LoadingButton>
      </NBox>
    </ScreenContainer>
  );
}

export function EditDIDScreenContainer() {
  return <EditDIDScreen />;
}
