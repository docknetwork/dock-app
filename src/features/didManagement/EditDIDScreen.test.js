import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {EditDIDScreen} from './EditDIDScreen';

const mockStore = configureMockStore();

describe('EditDIDScreen', () => {
  it('should render correctly ', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<EditDIDScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
