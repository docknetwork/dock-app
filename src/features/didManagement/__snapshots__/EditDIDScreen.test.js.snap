// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditDIDScreen should render correctly  1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
  testID="CreateNewDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={1}
          marginRight={22}
        >
          <Memo(ForwardRef(Box))
            width="80px"
          >
            <BackButton
              onPress={[Function]}
            />
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignContent="center"
            alignItems="center"
            flex={1}
            pl={15}
            width="100%"
          >
            <Styled(Component)
              variant="h3"
            >
              Edit DID
            </Styled(Component)>
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignItems="flex-end"
            width="80px"
          />
        </Box>
      </Styled(View)>
      <ForwardRef
        marginLeft={5}
        marginRight={5}
      >
        <Memo(ForwardRef(FormControl))>
          <Memo(ForwardRef(Stack))
            mt={7}
          >
            <Memo(ForwardRef(FormControlLabel))>
              DID Name
            </Memo(ForwardRef(FormControlLabel))>
            <Memo(ForwardRef(Input))
              autoCapitalize="none"
              value=""
            />
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(FormControl))>
      </ForwardRef>
      <Memo(ForwardRef(Box))
        mx={7}
      >
        <LoadingButton
          full={true}
          isDisabled={false}
          mb={70}
          onPress={null}
          testID="save-btn"
        >
          Save
        </LoadingButton>
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
